ccm_web/server/src/api/api.controller.ts

```diff
@@ -68,6 +69,32 @@ export class APIController {
     return result
   }
 
+  @UseInterceptors(InvalidTokenInterceptor)
+  @Get('instructor/sections')
+  async getCourseSectionsInTermAsInstructor (
+    @UserDec() user: User, @Query('term_id', ParseIntPipe) termId: number
+  ): Promise<CourseWithSections[]> {
+    const result = await this.apiService.getCourseSectionsInTermAsInstructor(user, termId)
+    if (isAPIErrorData(result)) throw new HttpException(result, result.statusCode)
+    return result
+  }
+
+  @UseInterceptors(InvalidTokenInterceptor)
+  @ApiQuery({ name: 'term_id', type: Number })
+  @ApiQuery({ name: 'instructor', required: false, type: String })
+  @ApiQuery({ name: 'course_name', required: false, type: String })
+  @Get('admin/sections')
+  async getCourseSectionsInTermAsAdmin (
+    @UserDec() user: User,
+      @Query('term_id', ParseIntPipe) termId: number,
```

Comment:
I don't think the UI know which terms this course belongs to. Term info mostly come into play Merge section feature. Either you have to send that info to FE through globals or store it into session or both.

Reply:
I'm sending it through the course endpoint (see changes to `/course/:id` endpoint and `CanvasCourseBase`, which the frontend gets on load and can pass to the feature. It seemed better to me to get it that way, but I could do through the launch/session.

Reply:
I *think* doing it this way may circumvent some of the issues related to #113, because the React application should still have in local state which course was sent in `globals` after the launch, and then be able to pull the latest term info even if session data changed.

Reply:
That would be yet another Prop along with global props. I see your point since the course API call is made anyways after the global call and the term data is present there. If you go through Globals I think it would be making API call to get the course term via course API endpoint and then repeat. 

What you did make sense!